cep-XXXX.md

```diff
@@ -0,0 +1,232 @@
+versioned `v1` and `v2` respectively.
+
+#### Distribution strings
+
+A "distribution string" MAY be used to identify a package artifact, without specifying the
+extension or channel. It MUST match the following syntax:
+
+```text
+[<subdir>/]<package name>-<version string>-<build string>
+```
+
+Distribution strings apply to distributable packages. They are used as the name of
+the directories where artifacts are extracted in the package cache, for example.
+
+Virtual packages MAY be also identified by a distribution string, but in those cases a subdir MUST NOT be present.
+
+> Note: Despite the similarity, distribution strings are not `MatchSpec`-like specifiers and MUST
+> NOT be used as such.
+
+#### Filenames
+
+The filename of distributable conda artifacts is obtained by adding the artifact extension to its
+distribution string (without the subdir, if present). It MUST match this syntax:
+
+```text
+<package name>-<version string>-<build string>.<extension>
+```
+
+The maximum length of a filename MUST NOT exceed 211 characters.
+
+Virtual conda packages do not exist on disk and SHOULD NOT need filename standardization.
+
+### Identifying channels
+
+A conda channel is defined as a URL where one can find one or more `repodata.json` files arranged
+in one subdirectory (_subdir_) each. `noarch/repodata.json` MUST be present to consider the parent
+location a channel.
+
+#### Channel base URLs
+
+The base URL for the arbitrary location of a repodata file is defined as:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]/<subdir>/repodata.json
+```
+
+with `<scheme>`, `<authority>` and `<path>` defined by [RFC
+3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2).
+
+Taken the channel definition above, the base URL without trailing slashes is thus:
+
+```text
+<scheme>://[<authority>][/<path>/][/label/<label name>]
+```
+
+For example, given `https://conda.anaconda.org/conda-forge/noarch/repodata.json`, the part leading
+to `noarch/repodata.json` and thus base URL is `https://conda.anaconda.org/conda-forge`. For local
+repodata such as `file:///home/username/channel/noarch/repodata.json`, the channel base URL is
+`file:///home/username/channel`.
+
+When present, each path component MUST only contain lowercase ASCII letters, numbers, underscores,
+periods, and dashes. They MUST NOT start with a period or a dash. They SHOULD start and end with a
+letter or a number. If present, each path component MUST match this regex:
+
+```re
+^[a-z0-9_][a-z0-9_.-]*$
+```
+
+For `file://`-based channel URLs, the path component rules MAY be understood as recommendations
+only.
+
+The maximum length of an individual path component in a channel base URL MUST NOT exceed 128
+characters. The maximum length of a channel base URL SHOULD NOT exceed 256 characters.
+
+To avoid ambiguous `MatchSpec` grammar, the last path component of a channel base URL MUST NOT match any `subdir` identifiers.
```

Comment:
```suggestion
To avoid ambiguous `MatchSpec` grammar, the last path component of a channel base URL SHOULD NOT match any `subdir` identifiers. If it does, the behavior in this ambiguous
case is not defined and implementation dependent.
```